tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -139,6 +139,8 @@ def __init__(self, metadata: TestSuiteMetadata) -> None:
         self._lisa_environment_name: str  # Name assigned by LISA to the test environment, useful for correlation with LISA logs
         self._environment_name: str  # Name assigned by the AgentTestSuiteCombinator to the test environment
 
+        self._vmss_resource_group_count: int # Counter to keep track of how many resource groups have been created for vmss suites
```

Comment:
I added this counter because we need unique resource group names when running the test suite on all endorsed images

Reply:
makes sense. the counter should be local to the AgentTestSuite, though, don't you think? did you add it in the runbook to avoid concurrency issues when incrementing it? that can be handled with an rlock

Reply:
Not sure I understand, Is there only one instance AgentTestSuite or is there an instance per environment?

Reply:
There is one instance of AgentTestSuite per LISA environment (i.e. for each environment created by the AgentTestSuiteCombinator), all of them executing concurrently according to the 'concurrency' given in the runbook.

I posted my previous comment because I think the counter should be local to AgentTestSuite and I was wondering if you created in in the runbook/combinator due to concerns about concurrent executions of AgentTestSuite. 

Is there a different motivation to do that? If the concern was concurrency, that can be addressed with r locks, as in AgentTestSuite._create_working_directory, or AgentTestSuite._setup_test_run.

Reply:
If there is a new instance of the AgentTestSuite for each env, then wouldn't the counter be initialized to 0 each time?

Reply:
You would make it static (a class member, rather than an instance member), just as the rlocks in these 2 functions

Reply:
I mean, both the counter and its lock would be class members